### DISCUSSION TITLE
[HOWTO]how to export c function with pub access?

### DISCUSSION THREAD

**Discussion by Student414** [vlang/v#11934] [Ideas]
now i have made a v module called ffmpeg.
it's a v binding for ffmpeg, such as name.
but i have to export every functions with function wrapper, such as:
```
module avutil

// some c flags

fn C.avutil_xxx() int
pub fn avutil_xxx() int {
    return C.avutil_xxx()
}
```
it's so strange, and redundant.
so can i find a easy way to export c functions in module?

**Comment by irishgreencitrus** [comment]
this will probably be added in c2v in v 0.3.

**Comment by dumblob** [comment]
> it's so strange, and redundant.

I'd argue it's the other way around :wink:. Because either you want *V* API (in which case you'll wrap the C api in some higher-level *V*-specific concepts) or you want `1:1` API in which case you'll use `c2v` and won't bother any more :wink:.

Or you want both, but that doesn't change anything in what I wrote.
